internal/controller/article.go

```diff
@@ -0,0 +1,80 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/newUser1337/task-news/internal/entity"
+
+	"github.com/gorilla/mux"
+)
+
+type articleUsecase interface {
+	GetArticles(context.Context) ([]*entity.NewsletterNewsItemResponse, error)
+	GetArticle(context.Context, int64) (*entity.NewsArticleResponse, error)
+}
+
+type ArticleHandler struct {
+	article articleUsecase
+}
+
+func NewArticlesHandler(article articleUsecase) *ArticleHandler {
+	return &ArticleHandler{
+		article: article,
+	}
+}
+
+func (a *ArticleHandler) GetArticles(w http.ResponseWriter, req *http.Request) {
+	articles, err := a.article.GetArticles(req.Context())
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
you return error 500 but no error text explaining to a client what happened. Sounds like a bad practice. Usually there is a json response holding error message.

Reply:
fixed